factorybuildershandler.lua

```diff
@@ -60,25 +60,35 @@ function FactoryBuildersHandler:PrePositionFilter()
 		local isExperimental = expFactories[factoryName] or leadsToExpFactories[factoryName]
 		local mtype = factoryMobilities[factoryName][1]
 		if ai.needAdvanced and not ai.haveAdvFactory and not isAdvanced then
-			self:EchoDebug('not advanced when i need it')
+			self:EchoDebug(factoryName ..' not advanced when i need it')
 			buildMe = false 
 		end
 		if buildMe and ai.needExperimental and not ai.haveExpFactory and not isExperimental then
-			self:EchoDebug('not Experimental when i need it')
+			self:EchoDebug(factoryName ..' not Experimental when i need it')
 			buildMe = false 
 		end
 		if buildMe and (not ai.needExperimental or ai.haveExpFactory) and expFactories[factoryName] then
 			self:EchoDebug('Experimental when i dont need it')
 			buildMe = false 
 		end
 		if buildMe and mtype == 'air' and ai.factoryBuilded['air'][1] >= 1 and utn.needsWater then
-			self:EchoDebug('dont build seaplane if i have normal planes')
+			self:EchoDebug(factoryName .. ' dont build seaplane if i have normal planes')
 			buildMe = false 
 		end
 		if not buildMe and mtype == 'air' and ai.haveAdvFactory and ai.factoryBuilded['air'][1] > 0 and ai.factoryBuilded['air'][1] < 3 and isAdvanced then
-			self:EchoDebug('force build t2 air if you have t1 air and a t2 of another type')
+			self:EchoDebug(factoryName .. ' force build t2 air if you have t1 air and a t2 of another type')
 			buildMe = true
 		end
+		if buildMe and self.ai.factoriesAtLevel[1] and mtype == 'air' and isAdvanced and not ai.haveAdvFactory then
+			for index, factory in pairs(self.ai.factoriesAtLevel[1]) do
+				if unitTable[factory.unit:Internal():Name()].mtype ~= 'air' then
```

Comment:
https://github.com/eronoobos/BABAR-The-Shardifant/commit/e4fdd6a825e6509ca2957205700b56a7e027480f